cli/src/cli_util.rs

```diff
@@ -2658,21 +2658,27 @@ fn resolve_default_command(
 
     if let Some(matches) = matches {
         if matches.subcommand_name().is_none() {
-            if config.get_string("ui.default-command").is_err() {
-                writeln!(
-                    ui.hint(),
-                    "Hint: Use `jj -h` for a list of available commands."
-                )?;
-                writeln!(
-                    ui.hint(),
-                    "Set the config `ui.default-command = \"log\"` to disable this message."
-                )?;
+            let mut default_command = match config.get("ui.default-command") {
+                Err(_) | Ok(CommandNameAndArgs::Structured { .. }) => {
+                    writeln!(
+                        ui.hint(),
+                        "Hint: Use `jj -h` for a list of available commands."
+                    )?;
+                    writeln!(
+                        ui.hint(),
+                        "Set the config `ui.default-command = \"log\"` to disable this message."
+                    )?;
+
+                    vec!["log".to_string()]
+                }
+                Ok(CommandNameAndArgs::String(name)) => vec![name],
+                Ok(CommandNameAndArgs::Vec(v)) => v.into(),
+            };
+            default_command.reverse();
+            for arg in default_command {
+                // Insert the default command directly after the path to the binary.
+                string_args.insert(1, arg);
             }
```

Comment:
Maybe instead of reversing and then repeatedly inserting at position 1 we could just `Vec::splice` `default_command` into `string_args` at range `1 + default_command.len()`?

There may be some ownership/lifetime thing at play here; I can't typecheck the code in my head right now. :)

Reply:
I think you want [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8f42133e186b75124f12e0cf956d2f8e) right?

I've implemented in a new commit :ribbon:

Reply:
That looks good, thank you!

Note that I think we would generally want the commits to be merged together, since the second commit just addresses a review comment. We don't squash commits when we merge them in the GitHub UI; we rebase every commit onto master, and we review commit-by-commit.

You can do that with `jj move --from A --to B` to move all the changes from one commit to the other and then re-push. It's a more general version of `jj squash` and `jj unsquash`; I use it a lot.

Reply:
Oh, and make sure that after you squash those two together, you make sure the branch name you set (`jj branch set`) is pointed to the correct, final commit! `jj branch set allow-list-ui-default-command -r ...` should be right, if you give it the correct revision. Then you can simply do `jj git push`

Reply:
Ok no problem! I created the new commit just to ease rolling back, but I imagined I would have to put them together.